portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesUseCase.testcase

```diff
@@ -81,6 +82,170 @@ definition {
+
+			JSONUser.setFirstPassword(
+				agreeToTermsAndAnswerReminderQuery = "true",
+				requireReset = "false",
+				userEmailAddress = "userea@liferay.com");
+
+			JSONUser.addUserToSite(
+				groupName = "Guest",
+				userEmailAddress = "userea@liferay.com");
+
+			CookiePreferenceHandlingAdmin.disablePreferenceHandlingWithOutCookieConsentInSiteSettings(
+				baseURL = "http://localhost:8080",
+				portlet = "Preference Handling",
+				siteURLKey = "guest");
+
+			CookiePreferenceHandlingAdmin.disablePreferenceHandling(explicitMode = "disable");
+		}
+
+		task ("When: End User first visits the site") {
+			UserBar.signOut();
+
+			User.firstLoginUI(
+				password = "test",
+				specificURL = "http://localhost:8080",
+				userEmailAddress = "userea@liferay.com");
+		}
+
+		task ("Then: Verify No cookies are set, only the functional cookie: GUEST_LANGUAGE_ID is set verify cookie banner is not present.") {
+			CookieBanner.verifyCookieBannerIsNotPresent();
+
+			CookieConsentPanel.verifyCookiesAreNotPresent();
+
+			CookieHelper._checkSelectedCookieValue(
+				cookieName = "GUEST_LANGUAGE_ID",
+				expectedCookieValue = "en_US");
+		}
+	}
+
+	@description = "This is a use case for LPS-170904 TC-6. Verify only functional cookies are set in instance level and Implicit mode, when the user is a first-time visitor."
+	@priority = 5
+	test VerifyFunctionalCookiesForFirstVisitInstanceLevelImplicitMode {
+		property test.name.skip.portal.instance = "CookiesSiteLevelUseCase#VerifyFunctionalCookiesForFirstVisitInstanceLevelImplicitMode";
+
+		task ("Given: Site Administrator enabled the cookie preference handling in instance level and the explicit cookie consent mode.") {
+			JSONUser.addUser(
+				userEmailAddress = "userea@liferay.com",
+				userFirstName = "userfn",
+				userLastName = "userln",
+				userScreenName = "usersn");
+
+			JSONUser.setFirstPassword(
+				agreeToTermsAndAnswerReminderQuery = "true",
+				requireReset = "false",
+				userEmailAddress = "userea@liferay.com");
+
+			JSONUser.addUserToSite(
+				groupName = "Guest",
+				userEmailAddress = "userea@liferay.com");
+
+			CookiePreferenceHandlingAdmin.disablePreferenceHandling();
+
+			CookiePreferenceHandlingAdmin.disablePreferenceHandlingWithOutCookieConsentInSiteSettings(
+				baseURL = "http://localhost:8080",
+				portlet = "Preference Handling",
+				siteURLKey = "guest");
+
+			CookiePreferenceHandlingAdmin.enablePreferenceHandlingWithOutCookieConsentInInstanceSittings();
+		}
+
+		task ("When: End User first visits the site") {
+			UserBar.signOut();
+
+			User.firstLoginUI(
+				password = "test",
+				specificURL = "http://localhost:8080",
+				userEmailAddress = "userea@liferay.com");
+		}
+
+		task ("Then: Verify No cookies are set, not even the functional cookie: GUEST_LANGUAGE_ID verify cookie banner is present.") {
+			CookieBanner.verifyCookieBannerIsPresent();
+
+			CookieConsentPanel.verifyCookiesAreNotPresent();
+
+			CookieHelper._checkSelectedCookieValue(
```

Comment:
@melodywu We should extend this macro with the cookies' domain check also. In this case, it's important to know that the cookie's domain is matching with the Instance's domain eg.: localhost

Reply:
Yes, I need to extend the macro. Could you teach me how to get domain value from cookie table use javascript? Like CookieHelper._checkSelectedCookieValuemacro 

		```
var javascript = '''var cookieName = "${cookieName}";
                        	var allCookies = document.cookie.split(';');
							var value="empty";
                        	for (var i = 0; i < allCookies.length; i++){
                        		if(allCookies[i].includes(cookieName)){
                        			value = allCookies[i].replace(cookieName+"=","").trim();
                        		}
                        	}
                        	return value;''';
```